cas-server-core-events-jpa/src/main/java/org/jasig/cas/config/JpaEventsConfiguration.java

```diff
@@ -0,0 +1,232 @@
+public class JpaEventsConfiguration {
+
+    /**
+     * The Show sql.
+     */
+    @Value("${database.show.sql:true}")
+    private boolean showSql;
+
+    /**
+     * The Generate ddl.
+     */
+    @Value("${database.gen.ddl:true}")
+    private boolean generateDdl;
+
+    /**
+     * The Hibernate dialect.
+     */
+    @Value("${events.jpa.database.dialect:org.hibernate.dialect.HSQLDialect}")
+    private String hibernateDialect;
+
+    /**
+     * The Hibernate hbm 2 ddl auto.
+     */
+    @Value("${events.jpa.database.ddl.auto:create-drop}")
+    private String hibernateHbm2DdlAuto;
+
+    /**
+     * The Hibernate batch size.
+     */
+    @Value("${events.jpa.database.batchSize:1}")
+    private String hibernateBatchSize;
+
+    /**
+     * The Driver class.
+     */
+    @Value("${events.jpa.database.driverClass:org.hsqldb.jdbcDriver}")
+    private String driverClass;
+
+    /**
+     * The Jdbc url.
+     */
+    @Value("${events.jpa.database.url:jdbc:hsqldb:mem:cas-events-registry}")
+    private String jdbcUrl;
+
+    /**
+     * The User.
+     */
+    @Value("${events.jpa.database.user:sa}")
+    private String user;
+
+    /**
+     * The Password.
+     */
+    @Value("${events.jpa.database.password:}")
+    private String password;
+
+    /**
+     * The Initial pool size.
+     */
+    @Value("${events.jpa.database.pool.minSize:6}")
+    private int initialPoolSize;
+
+    /**
+     * The Min pool size.
+     */
+    @Value("${events.jpa.database.pool.minSize:6}")
+    private int minPoolSize;
+
+    /**
+     * The Max pool size.
+     */
+    @Value("${events.jpa.database.pool.maxSize:18}")
+    private int maxPoolSize;
+
+    /**
+     * The Max idle time excess connections.
+     */
+    @Value("${events.jpa.database.pool.maxIdleTime:1000}")
+    private int maxIdleTimeExcessConnections;
+
+    /**
+     * The Checkout timeout.
+     */
+    @Value("${events.jpa.database.pool.maxWait:2000}")
+    private int checkoutTimeout;
+
+    /**
+     * The Acquire increment.
+     */
+    @Value("${events.jpa.database.pool.acquireIncrement:16}")
+    private int acquireIncrement;
+
+    /**
+     * The Acquire retry attempts.
+     */
+    @Value("${events.jpa.database.pool.acquireRetryAttempts:5}")
+    private int acquireRetryAttempts;
+
+    /**
+     * The Acquire retry delay.
+     */
+    @Value("${events.jpa.database.pool.acquireRetryDelay:2000}")
+    private int acquireRetryDelay;
+
+    /**
+     * The Idle connection test period.
+     */
+    @Value("${events.jpa.database.pool.idleConnectionTestPeriod:30}")
+    private int idleConnectionTestPeriod;
+
+    /**
+     * The Preferred test query.
+     */
+    @Value("${events.jpa.database.pool.connectionHealthQuery:select 1}")
+    private String preferredTestQuery;
+
+
+    /**
+     * Jpa event vendor adapter hibernate jpa vendor adapter.
+     *
+     * @return the hibernate jpa vendor adapter
+     */
+    @Bean(name = "jpaEventVendorAdapter")
```

Comment:
OK, this is just a nit pick, but I'd prefer not to explicitly define bean name via annotation value, but rather let the method name dictate the bean name. Let's adopt this style throughout java config classes